solr/solrj/src/java/org/apache/solr/common/util/SimpleOrderedMap.java

```diff
@@ -146,43 +126,21 @@ public T remove(final Object key) {
     return super.remove((String) key);
   }
 
-  /**
-   * Copies all of the mappings from the specified map to this map. These mappings will replace any
-   * mappings that this map had for any of the keys currently in the specified map.
-   *
-   * @param m mappings to be stored in this map
-   * @throws NullPointerException if the specified map is null
-   */
   @Override
   public void putAll(final Map<? extends String, ? extends T> m) {
     m.forEach(this::put);
```

Comment:
if you look at NamedList's shallow Map putAll impl, there's a faster implementation if the map is empty at the start.  We could incorporate that here with a simple check that calls `this::add`